### PR TITLE
validator: reject update request before syncing disk spec and status

### PR DESCRIPTION
**Pull request by derekbit**
The disk spec and status should be one-to-one matching.

Longhorn 685

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: webhook/resources/node/validator.go [hunk 1]
```diff
@@ -142,3 +142,17 @@ func (n *nodeValidator) Update(request *admission.Request, oldObj runtime.Object
...
 	return nil
 }
+
+func isNodeDiskSpecAndStatusSynced(node *longhorn.Node) bool {
+	if len(node.Spec.Disks) != len(node.Status.DiskStatus) {
+		return false
+	}
+
+	for diskName := range node.Spec.Disks {
+		if _, ok := node.Status.DiskStatus[diskName]; !ok {
```
Comment (shuo-wu) [inline webhook/resources/node/validator.go:152]: Need to check the path as well

**Review by shuo-wu — CHANGES_REQUESTED** [review]

File: webhook/resources/node/validator.go (hunk 1 above)
Comment (derekbit) [reply webhook/resources/node/validator.go:152]: `Path`? DiskStatus does not have `Path`.

**Review by innobead — APPROVED** [review]
LGTM

File: webhook/resources/node/validator.go (hunk 1 above)
Comment (shuo-wu) [reply webhook/resources/node/validator.go:152]: Oh, sorry. You are right

**Pull request merged**